src/shared/parameter_v2/examples/generated_dynamic_parameters.hpp

```diff
@@ -0,0 +1,351 @@
+#pragma once
+#include <iostream>
+#include <boost/program_options.hpp>
+#include "software/parameter/config.h"
+#include "software/parameter/numeric_parameter.h"
+#include "software/parameter/enumerated_parameter.h"
+
+class foo;
+class example;
+
+class foo : public Config
+{
+   public:
+    foo()
+    {
+        foo_bool = std::make_shared<Parameter<bool>>("foo_bool", true);
+        foo_int = std::make_shared<NumericParameter<int>>("foo_int", 3, 0, 5);
+        mutable_internal_param_list = {
+            foo_bool,
+            foo_int,
+        };
+        immutable_internal_param_list = {
+            std::const_pointer_cast<const Parameter<bool>>(foo_bool),
+            std::const_pointer_cast<const NumericParameter<int>>(foo_int),
+        };
+    }
+    
+    const std::shared_ptr<const Parameter<bool>> fooBool() const
+    {
```

Comment:
just checking that it isn't a style convention to name parameters using the type

Reply:
nope its not, the example params just have the type in the name